### DISCUSSION TITLE
Snippets not matched correctly

### DISCUSSION THREAD

**Discussion by medwatt** [L3MON4D3/LuaSnip#1172] [Q&A]
I have the snippets below. When I type `##` and expand the snippet, I expect to get `second`. However, I get `#first`. According to the docs, `wordTrig` is `true` by default, which means that the text before the cursor matches the trigger entirely before the snippet fires. So, I would expect `##` be detected completely.

```lua
s( { trig = "#", dscr }, { t("first") }),
s( { trig = "##", dscr }, { t("second") }),
``

**Comment by L3MON4D3** [comment]
Ah, we define word not as `[^ ]+`, but as `[%w_]+`, so neither `#` nor `##` are "words" :sweat_smile:
They expand nonetheless because wordTrig only makes sure that the character before the trigger is not in `[%w_]`, so the trigger may contain non-word characters, but expansion may not "split" a word.

`#` rather than `##` being expanded is just due to the order they are added (we use the first matching snippet), if you want a future-proof way of making sure that `##` is preferred, add `priority = 1001` after `trig="##"`, that way it will always be checked before `#` :)
